reverse_engineering/reFromFile.js

```diff
@@ -0,0 +1,146 @@
+const _ = require('lodash');
+const { initPluginConfiguration } = require('../shared/customProperties');
+const mapJsonSchema = require('../shared/mapJsonSchema');
+const convertToJsonSchemas = require('../reverse_engineering/helpers/convertToJsonSchemas');
+const { openAvroFile } = require('../reverse_engineering/helpers/fileHelper');
+const { getNamespace, handleErrorObject } = require('../reverse_engineering/helpers/generalHelper');
+
+const reFromFile = async (data, logger, callback, app) => {
+	initPluginConfiguration(data.pluginConfiguration, logger);
+
+	try {
+		const { filePath } = data;
+		const avroSchema = await openAvroFile(filePath);
+		const jsonSchemas = convertToJsonSchemas(avroSchema);
+
+		const { schemaRegistryType, schemaRegistryUrl } = _.first(getSchemasData(avroSchema));
+
+		return callback(
+			null,
+			getPackages(avroSchema, jsonSchemas),
+			{
+				schemaRegistryType,
+				schemaRegistryUrl,
+			},
+			[],
+			'multipleSchema',
+		);
+	} catch (err) {
+		const errorData = handleErrorObject(err);
+		logger.log('error', errorData, 'Parsing Avro Schema Error');
+
+		return callback(errorData);
+	}
+};
+
+const getPackages = (avroSchema, jsonSchemas) => {
+	const schemasData = getSchemasData(avroSchema);
+	const isAvroSchemaSplittedIntoMultipleJsonSchemas = schemasData.length === 1 && jsonSchemas.length > 1;
+	const singleNamespaceForOneAvroSchemaSplittedIntoMultipleJsonSchemas = schemasData[0]?.namespace;
+
+	return jsonSchemas.map((jsonSchema, index) => {
+		const { namespace, schemaType, schemaGroupName, confluentSubjectName, schemaTopic, confluentVersion } =
+			schemasData[index] || {};
+		const schemaNamespace = isAvroSchemaSplittedIntoMultipleJsonSchemas
+			? singleNamespaceForOneAvroSchemaSplittedIntoMultipleJsonSchemas
+			: namespace;
+
+		const schemaNameStrategy = inferSchemaNameStrategy({
+			name: jsonSchema.title,
+			namespace,
+			confluentSubjectName,
+			schemaTopic,
+		});
+		let references = [];
+		mapJsonSchema(field => {
```

Comment:
@Nightlngale I know it's not you but I find this function super convoluted. It's pretty hard to understand the side effects and what it really does (and i'm under the impression it does two things here...)

Reply:
The way we retrieve the references list is really weird tbh...